lib/Command/ExpirePassword.php

```diff
@@ -106,54 +130,105 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+		$uids = $input->getOption('uid');
 
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
-
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
+		$expireDate = new \DateTime();
+		$oldDate = new \DateTime();
+		$timeZone = new \DateTimeZone('UTC');
+		$inputExpireDate = $input->getArgument('expiredate');
+
+		//This array will hold the uids of users process when group option is passed
+		$users = [];
+		if ($allUsers !== false) {
+			$this->userManager->callForAllUsers(function (IUser $user) use ($expireDate, $oldDate, $timeZone, $inputExpireDate, $output, &$users) {
+				if (isset($users[$user->getUID()]) === false) {
+					if ($user->canChangePassword()) {
+						$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+						$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+						$users[$user->getUID()] = true;
+					}
+				}
+			});
+		} elseif (\count($groups) >= 1) {
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === true) {
+					foreach ($this->groupManager->findUsersInGroup($group) as $user) {
+						if (isset($users[$user->getUID()]) === false) {
+							if ($user->canChangePassword()) {
+								$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+								$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+								$users[$user->getUID()] = true;
+							}
+						}
+					}
+				} else {
+					$output->writeln("Ignoring missing group $group");
+				}
+			}
 		}
-
-		if (!$user->canChangePassword()) {
-			$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
-			return self::EX_GENERAL_ERROR;
+		if (\count($uids) > 0) {
```

Comment:
If both the `--all` and `-u` options are provided, we should ignore the `-u` options, which means this code shouldn't be executed in that scenario.